### PR TITLE
Accounts Cache for Verkle/Kaustinen

### PR DESCRIPTION
**Pull request by agnxsh**
Rewrite of accounts cache in favour of Verkle Specs, Notable changes:
1. No storage trie, as storage is now a part of the unified Verkle Trie (EIP 6800)
2. Changing the format of Witness Data and Witness Cache
3. Changing the trie in AccountsCache to a VerkleTrie
~~4. Changing the format of rootHash to a 32-byte array~~
5. Dealing with getter and setter functions related to contract code differently as EIP6800 introduces `chunkifyCode`
6. Dealing with getter and setter functions related to storage differently because of 1)
7. Added the checker function of whether the account is Empty or not
8. Added the persist code for AccountsCache
9. Added a placeholder for `deleteAccountBytes`, since deletion is kept like that as per spec and geth

Still a draft PR, not tested yet. Though underlying functions related to verkle-based accounts are tested. Verkle's account cache directly uses #2133

### CODE REVIEW

**Comment by advaita-saha** [comment]
The `rootHash()` return type was `KeccakHash` which has been kept intact to preserve the API and smoothen out future integration

**Comment by agnxsh** [comment]
IsEmptyVerkleAccount should be false in the getAccount check, only then the flag is Alive

**Comment by advaita-saha** [comment]
> @advaita-saha IsEmptyVerkleAccount should be false in the getAccount check, only then the flag is Alive

Ahh my bad

File: nimbus/db/ledger/verkle_accounts_cache.nim [hunk 1]
```diff
@@ -0,0 +1,530 @@
...
+proc persistCode(acc: RefAccount, address: EthAddress, db: VerkleTrie) =
+  if acc.code.len != 0:
+    when defined(geth):
+      VerkleTrieRef(db).updateContractCode(address, acc.account.codeHash, acc.code)
+    else:
+      VerkleTrieRef(db).updateContractCode(address, acc.account.codeHash, acc.code)
+
+proc persistStorage(acc: RefAccount, db: VerkleTrie, clearCache: bool) =
+  ## Placeholder for @agnxsh to implement using the verkle trie storage layer
+  return
```
Comment (advaita-saha) [inline nimbus/db/ledger/verkle_accounts_cache.nim:272]: Could you @agnxsh  please go ahead and implement this ?

Comment (agnxsh) [reply nimbus/db/ledger/verkle_accounts_cache.nim:272]: Sure

**Comment by advaita-saha** [comment]
Missing Items

- [ ] originalStorageValue
- [ ] setStorage
- [ ] clearStorage
- [ ] cachedStorage (iterator)
- [ ] update
- [ ] witnessData
- [ ] collectWitnessData

_**Lasty figure out the getContractCode()**_

**Comment by advaita-saha** [comment]
Note :-

`getContractCode` will need a seperate key-value db implementation

**Pull request merged**